File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/data/GeodatabaseNativeObject.java
package com.valentingrigorean.arcgis_maps_flutter.data;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.esri.arcgisruntime.data.Geodatabase;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.BaseNativeObject;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeHandler;
import com.valentingrigorean.arcgis_maps_flutter.loadable.LoadableNativeHandler;

import io.flutter.plugin.common.MethodChannel;

public class GeodatabaseNativeObject extends BaseNativeObject<Geodatabase> {

    public GeodatabaseNativeObject(String objectId, Geodatabase geodatabase) {
        super(objectId, geodatabase, new NativeHandler[]{
                new LoadableNativeHandler(geodatabase)
        });
    }

    @Override
    public void onMethodCall(@NonNull String method, @Nullable Object args, @NonNull MethodChannel.Result result) {
        switch (method) {
            case "geodatabase#close":
                getNativeObject().close();
                result.success(null);
                break;
            default:
                super.onMethodCall(method, args, result);
                break;
        }
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/data/GeodatabaseNativeObject.kt
package com.valentingrigorean.arcgis_maps_flutter.data

import com.arcgismaps.data.Geodatabase
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.BaseNativeObject
import com.valentingrigorean.arcgis_maps_flutter.loadable.LoadableNativeHandler
import io.flutter.plugin.common.MethodChannel

class GeodatabaseNativeObject(objectId: String, geodatabase: Geodatabase) :
    BaseNativeObject<Geodatabase>(
        objectId, geodatabase, arrayOf(
            LoadableNativeHandler(geodatabase)
        )
    ) {
    override fun onMethodCall(method: String, args: Any?, result: MethodChannel.Result) {
        when (method) {
            "geodatabase#close" -> {
                nativeObject.close()
                result.success(null)
            }

            else -> super.onMethodCall(method, args, result)
        }
    }
}